R/conda.R

```diff
@@ -373,6 +375,14 @@ conda_version <- function(conda = "auto") {
   system2(conda_bin, "--version", stdout = TRUE)
 }
 
+
+numeric_conda_version <- function(conda = "auto") {
+  v <- conda_version(conda)
+  v <- sub("(conda) ([0-9.]+)", "\\2", v)
+  numeric_version(v)
```

Comment:
Is there any possibility that `conda --version` could return something like `conda 4.2-alpha` or something similar here? Or are we safe assuming it will be a parsable numeric version?

Reply:
I tried to chase this down in the conda code base and it seems your intuition is right. As best as I can figure, the version string is (maybe always?) generated from the last git tag. Looking through the tags it seems that "{version}" can sometimes have a suffix like 'rc0', 'b0', 'a1', 'beta2, ... there doesn't appear to be systematic approach I can discern. Also, sometimes the version can sometimes be `"{version}.post{n_commits}+{commit_hash}"`, where {version} will be the last tag.

https://github.com/conda/conda/blob/c1579681d1468af3d1b4af3083bed33f8391e861/conda/_vendor/auxlib/packaging.py#L165
https://github.com/conda/conda/tags

The `"conda "` prefix however seems hardcoded and will probably always be there. 
https://github.com/conda/conda/blob/c1579681d1468af3d1b4af3083bed33f8391e861/conda/cli/conda_argparse.py#L36

I will try to make the numeric version parser more robust.